spec/registry.md

```diff
@@ -132,7 +134,9 @@ The following options and their values are required to be available on the funct
    -  `never`
 - `style`
   - `decimal` (default)
-  - `percent` (see [Percent Style](#percent-style) below)
+  - `percent` (see [Percent Formatting](#percent-formatting) below)
+  - `currency` (OPTIONAL, see [Currency Formatting](#currency-formatting) below)
+  - `unit` (OPTIONAL, see [Unit Formatting](#unit-formatting) below)
```

Comment:
Eh, that probably works for me too. What about percent formatting? Should that have its own function as well?

Reply:
I think percent is fine where it is? It's more like scientific or other presentational gorp for formatting a pure number (you never really see a `Percent` type walking about). I could be convinced otherwise, though.

Reply:
We do also apply this transform to percentage values: https://github.com/unicode-org/message-format-wg/blob/4ea56e4bc8d15dc15beae150a95fb5904c988926/spec/registry.md?plain=1#L220-L221

If there's reason enough to define `:unit` and `:currency`, it would make it much easier to also define `:percent`, and not allow any of these to be used as operands.

Reply:
The transformation for percents is multiplying a number/moving the radix. It doesn't require changing the type or any external data. We could add a per-mille formattnig too and @macchiati has proposed adding `offset`. These are all pure number things, which is why I'd keep them together with `:number`/`:integer`. 

Units and currencies have a host of specialized options and accept different types. That's why I'd separate them: they are only numbers because they _contain_ numeric values and want to select like numbers.

I wouldn't object to providing a `:percent` function. I don't think it is required that we do so, obviously. But maybe we do it for consistency?